nion/swift/ExportDialog.py

```diff
@@ -261,15 +261,29 @@ class UnitType(enum.Enum):
 
 
 class ExportSizeModel(Observable.Observable):
-    def __init__(self, display_item: DisplayItem.DisplayItem) -> None:
+    unit_to_str_map = {
+        UnitType.PIXELS: "pixels",
+        UnitType.INCHES: "inches",
+        UnitType.CENTIMETERS: "centimeters"
+    }
+
+    str_to_unit_map = {
+        "pixels": UnitType.PIXELS,
+        "inches": UnitType.INCHES,
+        "centimeters": UnitType.CENTIMETERS
+    }
+
+    def __init__(self, display_item: DisplayItem.DisplayItem, units_model: Model.PropertyModel[str]) -> None:
         super().__init__()
+        self.__units_model = units_model
         display_size = self.__calculate_display_size_in_pixels(display_item)
-        self.__last_input_value: float = display_size.width
         self.__aspect_ratio = display_size.width / display_size.height
-        self.__units = UnitType.PIXELS
         self.__float_to_string_converter = Converter.FloatToStringConverter()
         self.__primary_field = 'width'
+        units_str: str = str(self.__units_model.value)
```

Comment:
This should be `units_str = self.__units_model.value or str()` since `value` may be `None`. Also, you don't need the type specifier in the original code (or the new code) since the type checker can determine the type from context.